### PR TITLE
NavSatFix ROS2 converters

### PR DESCRIPTION
**Pull request by mikebauer**
## Description of change
Add ROS2 converters for NavSatFix.

## Guide to reproduce test results.
```bash
bazel test //resim/msg:navsat_from_ros2_test
bazel test //resim/msg:fuzz_helpers_test
```
## Checklist:
- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [x] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: resim/msg/fuzz_helpers.hh [hunk 1]
```diff
@@ -147,6 +149,40 @@ Detection3D random_element(TypeTag<Detection3D> /*unused*/, InOut<Rng> rng) {
...
+      NavSatFix::COVARIANCE_TYPE_KNOWN,
+  };
+
+  NavSatFix result;
+  result.mutable_header()->CopyFrom(random_element<Header>(rng));
+  result.set_status(
+      STATUSES.at(random_element<std::size_t>(rng) % STATUSES.size()));
+  result.set_latitude(random_element<double>(rng));
+  result.set_longitude(random_element<double>(rng));
+  result.set_altitude_m(random_element<double>(rng));
```
Comment (knowlsie) [inline resim/msg/fuzz_helpers.hh:173]: Clearly we should be using SE3::DIMS here. (Joke, please don't :laughing:)

File: resim/msg/fuzz_helpers.hh [hunk 2]
```diff
@@ -147,6 +149,40 @@ Detection3D random_element(TypeTag<Detection3D> /*unused*/, InOut<Rng> rng) {
...
+
+  constexpr int COV_DIM = 9;
+  for (int ii = 0; ii < COV_DIM; ++ii) {
+    // Absolutely not a valid covariance, but we only care about data
+    // preservation.
```
Comment (knowlsie) [inline resim/msg/fuzz_helpers.hh:178]: You could do the old $\Sigma = X^TX$ if you want to be all fancy here... (But as a side note, I actually think the root matrix of a positive semi-definite matrix is often more natural as a "type". Wish we could change conventions on these things.)

File: resim/msg/navsat.proto [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+syntax = "proto3";
+
+import "resim/msg/header.proto";
+
+package resim.msg;
+
+// These have the same meaining as in
```
Comment (knowlsie) [inline resim/msg/navsat.proto:13]:
```suggestion
// These have the same meaning as in
```

File: resim/msg/navsat.proto [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        COVARIANCE_TYPE_UNKNOWN        = 0;
+        COVARIANCE_TYPE_APPROXIMATED   = 1;
+        COVARIANCE_TYPE_DIAGONAL_KNOWN = 2;
+        COVARIANCE_TYPE_KNOWN          = 3;
+    }
+
+    Header          header                   = 1;
+    Status          status                   = 2;
+    double          latitude                 = 3;
+    double          longitude                = 4;
```
Comment (knowlsie) [inline resim/msg/navsat.proto:32]: As a general comment, we may want to just quickly think on how to consistently label units before landing these. I'd argue `latitude_deg` might be good given `altitude_m` (and the fact that you need to convert latitudes to radians before doing the hyperbolic trig in the transverse mercator projection, so you do see them stored in radians sometimes...)

**Review by knowlsie — APPROVED** [review]
Sweet!

File: resim/msg/navsat.proto (hunk 3 above)
Comment (mikebauer) [reply resim/msg/navsat.proto]: Good catch. Fixed.

**Pull request merged**